### ISSUE TITLE
More sophisticated hyperparameter tuning

### ISSUE THREAD

**Issue by initze** [initze/thaw-slump-segmentation#26]
I just found this package, which might be helpful for optimization

https://github.com/autonomio/talos

**Comment by initze** [comment]
Optuna may also be good

https://github.com/optuna/optuna-examples/blob/main/pytorch/pytorch_simple.py

**Comment by initze** [comment]
Lerning rate scheduling, weems beneficial
https://pytorch.org/docs/stable/optim.html
* e.g. StepLR (https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.StepLR.html)
* exponential LR (ExponentialLR)

**Comment by khdlr** [comment]
We should definitely include that! I had already toyed around with (manual) LR Scheduling, but it's commented out (https://github.com/initze/thaw-slump-segmentation/blob/master/train.py#L192-L196). Exposing that as configurable, tunable options is a great idea.

People seem to get great results with Cosine schedules (https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingLR.html#torch.optim.lr_scheduler.CosineAnnealingLR). 

Also "warming up" the learning rate can help (https://nlp.seas.harvard.edu/2018/04/03/attention.html#optimizer), though that requires writing custom LR Schedulers. Happy to take care of that.

**Comment by initze** [comment]
deprecated for now
